actool/manifest.go

```diff
@@ -0,0 +1,364 @@
+		Description: `Copy an ACI and patch its manifest.`,
+		Summary:     "Copy an ACI and patch its manifest (experimental)",
+		Usage: `
+		  [--manifest=MANIFEST_FILE]
+		  [--name=example.com/app]
+                  [--capability=CAP_SYS_ADMIN,CAP_NET_ADMIN]
+		  [--mounts=work,path=/opt,readOnly=true[:work2,...]]
+		  [--replace]
+		  INPUT_ACI_FILE
+		  [OUTPUT_ACI_FILE]`,
+		Run: runPatchManifest,
+	}
+	cmdCatManifest = &Command{
+		Name:        "cat-manifest",
+		Description: `Print the manifest from an ACI.`,
+		Summary:     "Print the manifest from an ACI",
+		Usage:       ` [--pretty-print] ACI_FILE`,
+		Run:         runCatManifest,
+	}
+)
+
+func init() {
+	cmdPatchManifest.Flags.BoolVar(&patchOverwrite, "overwrite", false, "Overwrite target file if it already exists")
+	cmdPatchManifest.Flags.BoolVar(&patchNocompress, "no-compression", false, "Do not gzip-compress the produced ACI")
+	cmdPatchManifest.Flags.BoolVar(&patchReplace, "replace", false, "Replace the input file")
+
+	cmdPatchManifest.Flags.StringVar(&patchManifestFile, "manifest", "", "Manifest file")
+	cmdPatchManifest.Flags.StringVar(&patchName, "name", "", "Replace name")
+	cmdPatchManifest.Flags.StringVar(&patchCaps, "capability", "", "Replace capability")
+	cmdPatchManifest.Flags.StringVar(&patchMounts, "mounts", "", "Replace mount points")
+
+	cmdCatManifest.Flags.BoolVar(&catPrettyPrint, "pretty-print", false, "Print with better style")
+}
+
+func getIsolatorStr(name, value string) string {
+	return fmt.Sprintf(`
+                {
+                    "name": "%s",
+                    "value": { %s }
+                }`, name, value)
+}
+
+func patchManifest(im *schema.ImageManifest) error {
+
+	if patchName != "" {
+		name, err := types.NewACName(patchName)
+		if err != nil {
+			return err
+		}
+		im.Name = *name
+	}
+
+	if patchCaps != "" {
+		app := im.App
+		if app == nil {
+			return fmt.Errorf("no app in the manifest")
+		}
+		isolator := app.Isolators.GetByName(types.LinuxCapabilitiesRetainSetName)
+		if isolator != nil {
+			return fmt.Errorf("isolator already exists")
+		}
+
+		capsList := strings.Split(patchCaps, ",")
+		caps := fmt.Sprintf(`"set": ["%s"]`,
+			strings.Join(capsList, `", "`))
+		isolatorStr := getIsolatorStr(types.LinuxCapabilitiesRetainSetName,
+			caps)
+		isolator = &types.Isolator{}
+		err := isolator.UnmarshalJSON([]byte(isolatorStr))
+		if err != nil {
+			return err
+		}
+		app.Isolators = append(app.Isolators, *isolator)
+	}
+
+	if patchMounts != "" {
+		app := im.App
+		if app == nil {
+			return fmt.Errorf("no app in the manifest")
+		}
+		mounts := strings.Split(patchMounts, ":")
+		for _, m := range mounts {
+			mountPoint, err := types.MountPointFromString(m)
+			if err != nil {
+				return fmt.Errorf("cannot parse mount point %q", m)
+			}
+			app.MountPoints = append(app.MountPoints, *mountPoint)
+		}
+	}
+	return nil
+}
+
+// iterate iterates over the tar reader and optionally write in the tar writer if not nil
```

Comment:
can you call this extractManifest or something and update the docstring pls? "iterate" is a bit vague/misleading